### PR TITLE
fix(client): handle null values in BigInt columns

### PR DESCRIPTION
**Pull request by kevin-dp**
This PR fixes a bug where the Satellite process is trying to parse `null` values into a `BigInt`, as reported [on Discord](https://discord.com/channels/933657521581858818/1229458801799729233/1229458801799729233). I've extended a unit test to check BigInt values and also extended the e2e test for integers to check this corner case. The e2e test would reproduce the bug and confirms that it is solved by this fix.

### CODE REVIEW

File: clients/typescript/src/util/types.ts [hunk 1]
```diff
@@ -181,7 +181,7 @@ export function isDataChange(change: Change): change is DataChange {
 }
 
 export type Record = {
-  [key: string]: string | number | Uint8Array | undefined | null
+  [key: string]: string | number | bigint | Uint8Array | undefined | null
```
Comment (msfstef) [inline clients/typescript/src/util/types.ts:184]: is this necessary? I thought `Record` only contained potential serialised values ready to be fed to/returned by SQLite adapters, and bigint is always going to be serialized to a string, no?

**Review by msfstef — APPROVED** [review]
Good catch and nice work!

**Review by icehaunter — APPROVED** [review]

File: clients/typescript/src/util/types.ts (hunk 1 above)
Comment (kevin-dp) [reply clients/typescript/src/util/types.ts]: You're totally right! That is not possible with SQLite, only once we add PG but this is beyond the scope of this PR. Reverted these changes.
